[2024-05-03T12:22:04.337+0530] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-05-03T12:22:04.344+0530] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: async_dag.task scheduled__2024-05-03T06:50:00+00:00 [queued]>
[2024-05-03T12:22:04.346+0530] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: async_dag.task scheduled__2024-05-03T06:50:00+00:00 [queued]>
[2024-05-03T12:22:04.346+0530] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-05-03T12:22:04.354+0530] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): task> on 2024-05-03 06:50:00+00:00
[2024-05-03T12:22:04.356+0530] {standard_task_runner.py:63} INFO - Started process 25268 to run task
[2024-05-03T12:22:04.358+0530] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'async_dag', 'task', 'scheduled__2024-05-03T06:50:00+00:00', '--job-id', '338', '--raw', '--subdir', 'DAGS_FOLDER/async_dag.py', '--cfg-path', '/tmp/tmpzgbozelx']
[2024-05-03T12:22:04.358+0530] {standard_task_runner.py:91} INFO - Job 338: Subtask task
[2024-05-03T12:22:04.374+0530] {task_command.py:426} INFO - Running <TaskInstance: async_dag.task scheduled__2024-05-03T06:50:00+00:00 [running]> on host ninebit5lm
[2024-05-03T12:22:04.400+0530] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='bharat' AIRFLOW_CTX_DAG_ID='async_dag' AIRFLOW_CTX_TASK_ID='task' AIRFLOW_CTX_EXECUTION_DATE='2024-05-03T06:50:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-05-03T06:50:00+00:00'
[2024-05-03T12:22:04.401+0530] {taskinstance.py:430} INFO - ::endgroup::
[2024-05-03T12:22:04.401+0530] {python.py:237} INFO - Done. Returned value was: <coroutine object fn at 0x7fddb2603220>
[2024-05-03T12:22:04.401+0530] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-05-03T12:22:04.402+0530] {xcom.py:677} ERROR - Object of type coroutine is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your airflow config or make sure to decorate your object with attr.
[2024-05-03T12:22:04.402+0530] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/usr/local/lib/python3.10/dist-packages/airflow/serialization/serde.py", line 189, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'coroutine'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/airflow/models/taskinstance.py", line 486, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session)
  File "/usr/local/lib/python3.10/dist-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/airflow/models/taskinstance.py", line 3182, in xcom_push
    XCom.set(
  File "/usr/local/lib/python3.10/dist-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/airflow/models/xcom.py", line 247, in set
    value = cls.serialize_value(
  File "/usr/local/lib/python3.10/dist-packages/airflow/models/xcom.py", line 675, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/local/lib/python3.10/dist-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/local/lib/python3.10/dist-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type coroutine is not JSON serializable
[2024-05-03T12:22:04.404+0530] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=async_dag, task_id=task, execution_date=20240503T065000, start_date=20240503T065204, end_date=20240503T065204
[2024-05-03T12:22:04.410+0530] {standard_task_runner.py:110} ERROR - Failed to execute job 338 for task task (Object of type coroutine is not JSON serializable; 25268)
[2024-05-03T12:22:04.449+0530] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-05-03T12:22:04.458+0530] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-05-03T12:22:04.459+0530] {local_task_job_runner.py:222} INFO - ::endgroup::
